src/runtime/recipe/particle.ts

```diff
@@ -238,6 +253,26 @@ export class Particle {
     this.recipe.removeParticle(this);
   }
 
+  getSlotConnectionBySpec(spec: SlotSpec) {
+    return Object.values(this._consumedSlotConnections).find(slotConn => slotConn.getSlotSpec() === spec);
+  }
+
+  getSlotSpecByName(name: string) {
+    if (this.spec) {
```

Comment:
You can write `return this.spec && this.spec.slots.get(name);` if you like short. No pressure though.